lib/types.nix

```
@@ -275,6 +275,22 @@ rec {
       merge = mergeEqualOption;
     };
 
+    anyBool = mkOptionType {
```

Not sure I like `allBool`, which would be for the "and" behavior.
How about `enableBool`? It corresponds quite well to the typical "enable" scenario where `default = false;` and everyone just sets a default prio `true` if they need it, and duplicates merge happily.

The only name that's "high level" like `anyBool` I can think of now is this one. Maybe it's too high level.

```suggestion
    enableBool = mkOptionType {
```

If you don't like it, I think my preferred name would be the opposite of `boolAnd` (but not by a large margin). The pragmatic:

```suggestion
    boolOr = mkOptionType {
```

And to dump my last ideas...

The middle ground:

```suggestion
    boolByOr = mkOptionType {
```

or the explicit:

```suggestion
    boolMergedByOr = mkOptionType {
```

---

`boolOr` sounds too much like `nullOr`. People might expect `boolOr str` to be a boolean or a string.

I like `boolByOr` or `boolMergedByOr`